src/resources/elements/ConnectButton/ConnectButton.ts

```diff
@@ -58,7 +60,9 @@ export class ConnectButton {
       this.txPhase = Phase.None;
       this.txReceipt = null;
     }));
-
+    this.subscriptions.push(this.eventAggregator.subscribe("account.connect", async () => {
```

Comment:
why is this needed?

Reply:
When the user's wallet isn't connected, the "onConnect" method in the deals.ts code calls 

`this.eventAggregator.publish("account.connect");`

Which triggers the exact same code as the 'onConnect' method inside the ConnectButton.ts. The code you're referring to is needed to allow the deals.ts code to connect their wallet without having to duplicate code.